File: ruby-numerals/numerals_strings.rb
class NumeralsStrings


	def self.translate_0_to_9 num
		case num
		when 0
			'zero'
		when 1
			'one'
		when 2 
			'two'
		when 3
			'three'
		when 4
			'four'
		when 5
			'five'
		when 6
			'six'
		when 7
			'seven'
		when 8
			'eight'
		when 9
			'nine'
		end
	end

	def self.translate_10_to_19 num
		case num
		when 10
			'ten'
		when 11
			'eleven'
		when 12 
			'twelve'
		when 13
			'thirteen'
		when 14
			'fourteen'
		when 15
			'fifteen'
		when 16
			'sixteen'
		when 17
			'seventeen'
		when 18
			'eighteen'
		when 19
			'nineteen'
		end
	end

	def self.translate_20_to_99 num
		case num
		when 20..29
			'twenty'
		when 30..39
			'thirty'
		when 40..49
			'forty'
		when 50..59
			'fifty'
		when 60..69
			'sixty'
		when 70..79
			'seventy'
		when 80..89
			'eighty'
		when 90..99
			'ninety'
		end
	end

	def self.translate_x00
		'hundred'
	end

	def self.translate_x000
		'thousand'
	end
end

File: js-numerals/numeralsStrings.js
function translate0to9(num) {
	var translation = '';
	switch (parseInt(num)) {
		case 0: translation = 'zero'; break;
		case 1: translation = 'one'; break;
		case 2: translation = 'two'; break;
		case 3: translation = 'three'; break;
		case 4: translation = 'four'; break;
		case 5: translation = 'five'; break;
		case 6: translation = 'six'; break;
		case 7: translation = 'seven'; break;
		case 8: translation = 'eight'; break;
		case 9: translation = 'nine'; break;
	}
	return translation;
}

function translate10to19(num) {
	var translation = '';
	switch (parseInt(num)) {
		case 10: translation = 'ten'; break;
		case 11: translation = 'eleven'; break;
		case 12: translation = 'twelve'; break;
		case 13: translation = 'thirteen'; break;
		case 14: translation = 'fourteen'; break;
		case 15: translation = 'fifteen'; break;
		case 16: translation = 'sixteen'; break;
		case 17: translation = 'seventeen'; break;
		case 18: translation = 'eighteen'; break;
		case 19: translation = 'nineteen'; break;
	}
	return translation;
}

function translate20to99(num) {
	num = parseInt(num);
	var translation = '';
	if (num >= 20 && num <= 29) translation = 'twenty';
	if (num >= 30 && num <= 39) translation = 'thirty';
	if (num >= 40 && num <= 49) translation = 'forty';
	if (num >= 50 && num <= 59) translation = 'fifty';
	if (num >= 60 && num <= 69) translation = 'sixty';
	if (num >= 70 && num <= 79) translation = 'seventy';
	if (num >= 80 && num <= 89) translation = 'eighty';
	if (num >= 90 && num <= 99) translation = 'ninety';
	return translation;
}

function translateX00(num) {
	return 'hundred'
}

function translateX000(num) {
	return 'thousand'
}
